### PR TITLE
Created commit feature for Addressbook. So history of Addressbook will be recorded

### PR DESCRIPTION
**Pull request by adam07018**

### CODE REVIEW

**Review by YZmunchmunch — APPROVED** [review]
Good use of OOP, simple and clear method of implementing undo function. LGTM! I approve this code.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -96,21 +99,33 @@ public boolean hasPerson(Person person) {
...
+        commit();
+    }
+
+    //=========== Versioned Address Book =====================================================================
+    public void commit() {
+        versionedAddressBook.commit(addressBook);
     }
 
+    //todo
+    public void undo(){}
```
Comment (tituswe) [inline src/main/java/seedu/address/model/ModelManager.java:125]: Coding standard: whitespace between ")" and "{"

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -96,21 +99,33 @@ public boolean hasPerson(Person person) {
...
+
+    public void redo(){}
```
Comment (tituswe) [inline src/main/java/seedu/address/model/ModelManager.java:127]: Coding standard: whitespace between ")" and "{"

**Review by tituswe — APPROVED** [review]
LGTM. Just the one bug, can fix after merge.

File: src/main/java/seedu/address/model/VersionedAddressBook.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     * Move currentStatePointer one before
+     */
+    public void undo() {
+        currentStatePointer--;
+    }
+
+    //todo
+    public void redo() {
```
Comment (pangrwa) [inline src/main/java/seedu/address/model/VersionedAddressBook.java:49]: Do remember to keep us in the loop!

File: src/main/java/seedu/address/model/VersionedAddressBook.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    /**
+     * Insert node (address book) after currentStatePointer
+     * 1. Delete the following node at currentStatePointer
+     * 2. Add node
+     * 3. Reset currentStatePointer
+     * @param ab
+     */
+    public void commit(AddressBook ab) {
```
Comment (anunayajoshi) [inline src/main/java/seedu/address/model/VersionedAddressBook.java:30]: Thanks for using comments to make the code easier to understand!

**Review by anunayajoshi — APPROVED** [review]
LGTM! Good use of comments!

File: src/main/java/seedu/address/logic/commands/UndoCommand.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * Undo the previous command by tracing back to previous AddressBook
+ */
+public class UndoCommand extends Command {
+    public static final String COMMAND_WORD = "undo";
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        return new CommandResult("Hello from undo");
```
[the remark this answers is not in the document]
Comment (adam07018) [reply src/main/java/seedu/address/logic/commands/UndoCommand.java:14]: Hi, this is for testing purpose to check if the skeleton of. Undo Command is working properly

**Comment by adam07018** [comment]
@pangrwa @tituswe pr ready for merge

**Comment by adam07018** [comment]
Close out of date pull request

**Pull request closed without merging**